stripe/_util.py

```diff
@@ -25,16 +25,18 @@
 
 # Used for global variables
 import stripe  # noqa: IMP101
-from stripe._stripe_response import StripeResponse
-from stripe._stripe_object import StripeObject
+
+if TYPE_CHECKING:
+    from stripe._stripe_response import StripeResponse
+    from stripe._stripe_object import StripeObject
 
 STRIPE_LOG = os.environ.get("STRIPE_LOG")
 
 logger: logging.Logger = logging.getLogger("stripe")
 
-if hasattr(typing_extensions, "deprecated"):
```

Comment:
This `hasattr(` thing actually prevents proper typechecking. `if TYPE_CHECKING` is enough to prevent runtime crashes, and it turns out that even in cases where `typing_extensions` is too old, Typeshed will still define it and allow the type checker to work.